### PR TITLE
Sync with upstream@v0.5.2-93-gc051bfe4

### PR DESCRIPTION
**Pull request by rafvasq**
Merge [vllm-project/vllm](https://github.com/vllm-project/vllm):[main](https://github.com/vllm-project/vllm/tree/main)@[v0.5.2-93-gc051bfe4](https://github.com/vllm-project/vllm/commit/c051bfe4eb77b82eba90504360bbd4e61d9e489a) into main

### CODE REVIEW

**Comment by prashantgupta24** [comment]
Looks good @rafvasq ! Once we sync with upstream, we can get this in!

File: vllm/entrypoints/openai/api_server.py
Comment (prashantgupta24) [inline vllm/entrypoints/openai/api_server.py:1]: we don't need changes to this file anymore, the `__main__.py` file handles starting both servers now!

File: vllm/tgis_utils/hub.py [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+            preferred = preferred_names.intersection(complete_names)
+            if preferred:
+                keep_name = sorted(list(preferred))[0]
+        for name in sorted(shared):
+            if name != keep_name:
+                to_remove[keep_name].append(name)
+    return to_remove
+
+
+def convert_file(pt_file: Path, sf_file: Path, discard_names: List[str]):
```
Comment (prashantgupta24) [inline vllm/tgis_utils/hub.py:186]: Worth checking if this function is the same thing? - https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/model_loader/weight_utils.py#L81

File: vllm/scripts.py [hunk 2]
```diff
@@ -82,6 +96,151 @@ def chat(system_prompt: Optional[str], model_name: str,
         print(output)
 
 
+def download_weights(
```
Comment (prashantgupta24) [inline vllm/scripts.py:99]: I wonder if moving all these functions to a file that's not going to be changed by upstream vllm will help in future merge conflicts

**Review by dtrifiro — COMMENTED** [review]
Shouldn't all of this go to https://github.com/opendatahub-io/vllm-tgis-adapter ? @prashantgupta24

**Comment by prashantgupta24** [comment]
@dtrifiro I had debated that, but yeah that might make the future upstream conflict issue go away if we do that. Might need some changes to make it work with the adapter

**Comment by prashantgupta24** [comment]
@rafvasq can you try incorporating this in the vllm-tgis-adapter that Daniele linked above? Apologies for moving this PR around so much, but a lot of things are changing rapidly

**Comment by rafvasq** [comment]
No problem @prashantgupta24!

**Comment by dtrifiro** [comment]
Closing as this belongs in `vllm-tgis-adapter`.

**Pull request closed without merging**